### PR TITLE
EdgeAgent: Add structure around handling methods

### PR DESCRIPTION
**Pull request by varunpuranik**
Currently EdgeAgent supports only one simple method - "ping". But we have plans to add several more, for getting logs, uploading logs, plus some other helper ones. 
This PR adds support to cleanly add support for more such methods.

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/IRequestManager.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Requests
+{
+    using System.Threading.Tasks;
+
+    public interface IRequestManager
+    {
+        Task<(int, string)> ProcessRequest(string request, string payloadJson);
```
Comment (myagley) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/IRequestManager.cs]: Is it possible to create a data object for this return instead of the tuple. That way we can document the purpose of the fields.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/RequestManager.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            this.requestHandlers = Preconditions.CheckNotNull(requestHandlers, nameof(requestHandlers));
+        }
+
+        public async Task<(int, string)> ProcessRequest(string request, string payloadJson)
+        {
+            try
+            {
+                Preconditions.CheckNonWhiteSpace(request, nameof(request));
+                Events.HandlingRequest(request, payloadJson);
+                if (!this.requestHandlers.TryGetValue(request, out IRequestHandler requestHandler))
```
Comment (myagley) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/RequestManager.cs]: Should we make the keys case-insensitive?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/IRequestManager.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/IRequestManager.cs]: I added variables to the types.. that should convey the intent.

File: edgelet/build/windows/iotedge.wm.xml [hunk 3]
```diff
@@ -1,75 +1,75 @@
-<?xml version='1.0' encoding='utf-8' standalone='yes'?>
-<identity
```
Comment (varunpuranik) [inline edgelet/build/windows/iotedge.wm.xml]: Not sure why these changes are showing up here..
